Studies/python/BayesianOptimizationCustom.py

```diff
@@ -52,19 +60,29 @@ def TransformParams(self, params):
             else:
                 raise Exception('type {} not allowed'.format(self.params_typed_range[key]['type']))
 
+
         for key, value in params.items():
-            #Has some dependency wiht other variables
-            if 'depends_on' in self.params_typed_range[key]:
-                # the variable on which it depends has a value of zero
-                if new_params[self.params_typed_range[key]['depends_on'] ] == 0:
-                    new_params[key] = self.params_typed_range[key]['range'][0]
+            if type(self.params_typed_range[key]['range']) == list:
+                #Has some dependency wiht other variables
+                if 'depends_on' in self.params_typed_range[key]:
+                    # the variable on which it depends has a value of zero
+                    if new_params[self.params_typed_range[key]['depends_on'] ] == 0:
+                        new_params[key] = self.params_typed_range[key]['range'][0]
+
+        new_params.update(self.const_params)
+
         return new_params
 
     def InverseTransformParams(self, params):
+        #Copy of params, including constant parameters
         new_params = params.copy()
         for key, values in params.items():
-            if self.params_typed_range[key]['type'] == 'list':
-                new_params[key] = self.params_typed_range[key]['range'].index(values)
+             if key not in self.const_params:
+                 if self.params_typed_range[key]['type'] == 'list':
+                     new_params[key] = self.params_typed_range[key]['range'].index(values)
+             else:
+                 #Remove constant parameters from the opt space
+                 del new_params[key]
```

Comment:
better to avoid `del` operation. E.g. should be something like:
```python
def InverseTransformParams(self, params):
    new_params = {}
    for key, value in params.items():
        if key not in self.const_params:
            if self.params_typed_range[key]['type'] == 'list':
                new_params[key] = self.params_typed_range[key]['range'].index(value)
            else:
                new_params[key] = value
    return new_params
```